analyzers/tests/SonarAnalyzer.UnitTest/TestCases/NonDerivedPrivateClassesShouldBeSealed.CSharp11.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+
+file class ClassWontBeExtended { } // Noncompliant {{File-scoped classes which are not derived in the current assembly should be marked as 'sealed'.}}
+//         ^^^^^^^^^^^^^^^^^^^
+
+file record RecordWontBeExtended { } // Noncompliant {{File-scoped records which are not derived in the current assembly should be marked as 'sealed'.}}
+//          ^^^^^^^^^^^^^^^^^^^^
+
+file class FileClassVirtualMethod // Compliant, the class has a virtual member.
```

Comment:
I'm not sure I understand this case. Although the class has a virtual method, it is not derived. I would argue that the method should not be virtual. Maybe we should make a rule for that :).

Reply:
Made an [issue](https://github.com/SonarSource/sonar-dotnet/issues/7186)